src/openms/include/OpenMS/ANALYSIS/OPENSWATH/PeakPickerMobilogram.h

```diff
@@ -0,0 +1,166 @@
+    {
+    public:
+        //@{
+        /// Constructor
+        PeakPickerMobilogram();
+
+        /// Destructor
+        ~PeakPickerMobilogram() override {}
+        //@}
+
+        /// indices into FloatDataArrays of resulting picked mobilogram
+        enum FLOATINDICES { IDX_FWHM = 0, IDX_ABUNDANCE = 1, IDX_LEFTBORDER = 2, IDX_RIGHTBORDER = 3, SIZE_OF_FLOATINDICES };
+
+        /// Struct to hold peak positions
+        struct PeakPositions {
+            size_t left;    // Left position of the peak
+            size_t apex;    // Apex (highest point) position of the peak
+            size_t right;   // Right position of the peak
+        };
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram
+        */
+        void pickMobilogram(const Mobilogram& mobilogram, Mobilogram& picked_mobilogram);
+
+        /**
+            @brief Finds peaks in a single mobilogram and annotates left/right borders
+
+            It uses a modified algorithm of the PeakPickerHiRes
+
+            This function will return a picked mobilogram and a smoothed mobilogram
+        */
+        void pickMobilogram(Mobilogram mobilogram, Mobilogram& picked_mobilogram, Mobilogram& smoothed_mobilogram);
+
+        /**
+          @brief Helper function to find the highest peak in a mobilogram
+
+          This function takes the integrated intensities, left widths, right widths from the peak picker and finds the highest peak in the mobilogram. The left, apex, and right positions of the highest peak are returned.
+
+          If the peak picker found no peaks (i.e. intensities is empty), the returned PeakPositions object will have it's left, apex, and right set based on the input im_size. The peak picker could fail if the input mobilogram is sparse.
+
+          @note The left, apex, and right positions are indices into the mobilogram.
+
+          @param[in] intensities A vector of doubles containing the intensities of the peaks in the mobilogram
+          @param[in] left_widths A vector of Size values containing the left widths of the peaks in the mobilogram
+          @param[in] right_widths A vector of Size values containing the right widths of the peaks in the mobilogram
+          @param[in] im_size The size/length of the mobilogram
+          @return A PeakPositions object containing the left, apex, and right positions of the highest peak in the mobilogram
+        */
+        static PeakPositions findHighestPeak(const std::vector<double> intensities,
+                                       const std::vector<Size> left_widths,
+                                       const std::vector<Size> right_widths,
+                                       const size_t im_size);
+
+        /// Temporary vector to hold the integrated intensities
+        std::vector<double> integrated_intensities_;
```

Comment:
1. if REALLY needed those should be private
2. are these really needed ;)

Reply:
I think they are needed, because they're used by the main private `[pickMobilogram_](https://github.com/singjc/OpenMS/blob/a51888fe771b0d9f9858538a0f0420b194709f36/src/openms/source/ANALYSIS/OPENSWATH/PeakPickerMobilogram.cpp#L164-L169)` method?

They're also accessed publicly from the `[picker_](https://github.com/singjc/OpenMS/blob/a51888fe771b0d9f9858538a0f0420b194709f36/src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp#L560)` object, so I think they need to be public? But looking at this more, maybe I should use the actual `pickedMobilogram` object, and add the picked indices to the `FLOATINDICES`